### PR TITLE
Update group.py

### PR DESCRIPTION
**Pull request by ccl-core**
To fix bad-yield-annotation error

### CODE REVIEW

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py [hunk 1]
```diff
@@ -50,4 +50,4 @@ def __call__(self, *all_series: pd.DataFrame) -> pd.DataFrame:
             result: dict[str, Any] = {}
             for column in df.columns:
                 result[column] = row[column]
-            yield result
+            yield pd.DataFrame(result)
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py:53]:
```python
yield result
```

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def _map_column_name(self, column_name: str) -> str:
                 return field.name
         return column_name
 
-    def __call__(self, *all_series: pd.DataFrame) -> pd.DataFrame:
+    def __call__(self, *all_series: pd.DataFrame) -> Iterator[pd.DataFrame]:
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py:39]:
```python
Iterator[dict[str, Any]
```

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py (hunk 1 above)
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py:53]: bad return type [bad-return-type]

Expected: pandas.core.frame.DataFrame
Actually returned: Dict[str, Any]

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def _map_column_name(self, column_name: str) -> str:
                 return field.name
         return column_name
 
-    def __call__(self, *all_series: pd.DataFrame) -> pd.DataFrame:
+    def __call__(self, *all_series: pd.DataFrame) -> Iterator[Dict[str, Any]]:
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py:39]:
```python
Iterator[dict[str, Any]]
```

Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/group.py]: Done.

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ class ReadField(Operation):
 
     node: Field
 
-    def __call__(self, df: pd.DataFrame) -> pd.Series:
+    def __call__(self, df: pd.DataFrame) -> Any:
```
[the remark this answers is not in the document]
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:115]: Right, thanks! Done.

**Review by marcenacp — APPROVED** [review]

**Pull request merged**